### DISCUSSION TITLE
Flash problems

### DISCUSSION THREAD

**Discussion by PokoLoko2014** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#791] [Q&A]
Hello,

I just can't get it to flash the esp8266 node mcu v3. Could it be that some steps are not documented in the manual or does any file have to be edited beforehand?
I proceeded like this:

1.from the new platformio icon, open folder ‘code’.
2. connect USB data cable to ESP module.
3. build+upload firmware via USB. (By clicking on the Right arrow at bottom of screen)
(here I already get an error message)
4. build filesystem image
5. upload filesystem image

I also tested steps 4 and 5 before step 3, unfortunately without success.
Can someone please help me?

`Processing nodemcuv2 (platform: espressif8266@^4; board: nodemcuv2; framework: arduino)
--------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/nodemcuv2.html
PLATFORM: Espressif 8266 (4.0.1) > NodeMCU 1.0 (ESP-12E Module)
HARDWARE: ESP8266 160MHz, 80KB RAM, 4MB Flash
PACKAGES:
 - framework-arduinoespressif8266 @ 3.30002.0 (3.0.2)
 - tool-esptool @ 1.413.0 (4.13)
 - tool-esptoolpy @ 1.30000.201119 (3.0.0)
 - tool-mklittlefs @ 1.203.210628 (2.3)
 - tool-mkspiffs @ 1.200.0 (2.0)
 - toolchain-xtensa @ 2.100300.210717 (10.3.0)
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 47 compatible libraries
Scanning dependencies...
Dependency Graph
|-- ArduinoJson @ 6.21.2
|-- WebSockets @ 2.5.3
|-- PubSubClient @ 2.8.0
|-- ESPAsyncTCP @ 1.2.2
|-- EspSoftwareSerial @ 8.2.0
|-- DallasTemperature @ 3.11.0
|-- smarthomes
|-- BWC_unified
|-- ArduinoOTA @ 1.0
|-- ESP8266WebServer @ 1.0
|-- ESP8266WiFi @ 1.0
|-- LittleFS @ 0.1.0
|-- OneWire @ 2.3.8
|-- Ticker @ 1.0
Building in release mode
Compiling .pio\build\nodemcuv2\src\main.cpp.o
Compiling .pio\build\nodemcuv2\libb10\WebSockets\libb64\cencode.c.o
Compiling .pio\build\nodemcuv2\libb10\WebSockets\libsha1\libsha1.c.o
Compiling .pio\build\nodemcuv2\lib995\PubSubClient\PubSubClient.cpp.o
Compiling .pio\build\nodemcuv2\lib982\ghostl\ghostl\FastScheduler.cpp.o
Archiving .pio\build\nodemcuv2\libb10\libWebSockets.a
Compiling .pio\build\nodemcuv2\liba89\EspSoftwareSerial\SoftwareSerial.cpp.o
Archiving .pio\build\nodemcuv2\lib982\libghostl.a
Compiling .pio\build\nodemcuv2\libfbd\OneWire\OneWire.cpp.o
.pio\libdeps\nodemcuv2\PubSubClient\src\PubSubClient.cpp: In member function 'boolean PubSubClient::publish_P(const char*, const uint8_t*, unsigned int, boolean)':
.pio\libdeps\nodemcuv2\PubSubClient\src\PubSubClient.cpp:523:16: warning: comparison of integer expressions of different signedness: 'unsigned int' and 'int' [-Wsign-compare]
  523 |     return (rc == expectedLength);
      |             ~~~^~~~~~~~~~~~~~~~~
Archiving .pio\build\nodemcuv2\lib995\libPubSubClient.a
Archiving .pio\build\nodemcuv2\libfbd\libOneWire.a
Compiling .pio\build\nodemcuv2\lib00e\DallasTemperature\DallasTemperature.cpp.o
Compiling .pio\build\nodemcuv2\lib40c\cio\CIO_2021.cpp.o
Archiving .pio\build\nodemcuv2\liba89\libEspSoftwareSerial.a
src\main.cpp: In function 'void startNTP()':
src\main.cpp:430:80: error: no matching function for call to 'configTime(int, int, String&, const __FlashStringHelper*, const __FlashStringHelper*)'
  430 |     configTime(0,0,wifi_info->ip4NTP_str, F("pool.ntp.org"), F("time.nist.gov"));
      |                                                                                ^
In file included from src\main.h:2,
                 from src\main.cpp:1:
C:\Users\Dima & Kristina\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:269:6: note: candidate: 'void configTime(int, int, const char*, const char*, const char*)'
  269 | void configTime(int timezone, int daylightOffset_sec, const char* server1,
      |      ^~~~~~~~~~
C:\Users\Dima & Kristina\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:269:67: note:   no known conversion for argument 3 from 'String' to 'const char*'
  269 | void configTime(int timezone, int daylightOffset_sec, const char* server1,
      |                                                       ~~~~~~~~~~~~^~~~~~~
C:\Users\Dima & Kristina\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:272:6: note: candidate: 'void configTime(const char*, const char*, const char*, const char*)'
  272 | void configTime(const char* tz, const char* server1,
      |      ^~~~~~~~~~
C:\Users\Dima & Kristina\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:272:6: note:   candidate expects 4 arguments, 5 provided
Compiling .pio\build\nodemcuv2\lib40c\cio\CIO_2021HJT.cpp.o
Compiling .pio\build\nodemcuv2\lib40c\cio\CIO_4W.cpp.o
Archiving .pio\build\nodemcuv2\lib00e\libDallasTemperature.a
Compiling .pio\build\nodemcuv2\lib40c\cio\CIO_54149E.cpp.o
*** [.pio\build\nodemcuv2\src\main.cpp.o] Error 1
======================================== [FAILED] Took 14.76 seconds ========================================

 *  The terminal process "C:\Users\Dima & Kristina\.platformio\penv\Scripts\platformio.exe 'run', '--target', 'upload'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 
`

**Comment by visualapproach** [comment]
Would help if you told us what the error said

**Reply by PokoLoko2014** [reply]
Here is the code, sorry.

**Reply by visualapproach** [reply]
Interesting

**Reply by PokoLoko2014** [reply]
?

**Reply by visualapproach** [reply]
![image](https://github.com/user-attachments/assets/419f23bd-e054-47d7-8f78-f909b1a90fb4)

**Reply by PokoLoko2014** [reply]
I edit my first post, there is the code.

**Comment by visualapproach** [comment]
Found it! Didn't expect it here :)

**Comment by visualapproach** [comment]
Check that you have latest version of everything. This code and arduino framework etc.
Clean build from platformio icon menu.

**Comment by visualapproach** [comment]
Possibly the user name is the problem. Dina & Kristina contains special chars. Not sure but it can be a problem

**Reply by PokoLoko2014** [reply]
Thank you, finaly got it to work on another notebook 👍
The display reacts not properly, see the video:

https://drive.google.com/drive/folders/1bHGeo5KKxvskBlqHz01oydWPRftLYlkB

**Reply by visualapproach** [reply]
Have you set the right hardware config?

**Reply by PokoLoko2014** [reply]
![Uploading Screenshot_20240823_194051_Chrome.jpg…]()

**Reply by visualapproach** [reply]
Can't see anything in your post. Blank image.

**Comment by PokoLoko2014** [comment]
![Screenshot_20240823_194921_Chrome](https://github.com/user-attachments/assets/15ad31c7-6966-4d8a-887a-05c6a3c207dd)

**Comment by PokoLoko2014** [comment]
![Screenshot_20240823_201051_Chrome](https://github.com/user-attachments/assets/80c06870-473b-4c12-9ff1-665963c208ce)

Had to activate buttons 😅
The last problem are the temp buttons they make a sound but I cant set another temp, only via web ui.

**Comment by PokoLoko2014** [comment]
https://github.com/user-attachments/assets/fb67bfd6-de18-4cf8-9fb4-334949f956ff

The temperature adjustment definitely has a problem via home assistant (also web ui) the temp jumps back and forth wildly.
The buttons on the display the temp dont react, only hear the beeping.
